### PR TITLE
Adds form validation to admin settings system form

### LINKED ISSUE #192: Add form validation to admin/settings sytem form
**I'm submitting a ...**  (check one with "x")
- [ ] bug report
- [x] feature request
- [ ] support request => Please do not submit support requests here, instead ask your query in out Gitter channel at https://gitter.im/fossasia/open-event-frontend

image

### PR DESCRIPTION
**Pull request by CosmicCoder96**
#### Checklist

- [x] I have read the [Contribution & Best practices Guide](https://blog.fossasia.org/open-source-developer-guide-and-best-practices-at-fossasia).
- [x] My branch is up-to-date with the Upstream `development` branch.
- [x] The acceptance, integration, unit tests and linter pass locally with my changes 
- [x] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)

#### Short description of what this resolves:
Adds the form validation to `admin/settings/` system form

#### Changes proposed in this pull request:

image

Fixes #192

### CODE REVIEW

File: app/components/forms/admin/settings/system-form.js [hunk 1]
```diff
@@ -1,8 +1,186 @@
...
+              prompt : this.i18n.t('Please select a region')
+            }
+          ]
+        },
+
+        amazon_s3_bucket_name: {
+          identifier : 'amazon_s3_bucket_name',
+          rules      : [
+            {
+              type   : 'empty',
```
Comment (niranjan94) [inline app/components/forms/admin/settings/system-form.js]: For all this, also add `required=true` to the related form fields. for correct markup

File: app/components/forms/admin/settings/system-form.js [hunk 2]
```diff
@@ -1,8 +1,186 @@
...
+
+        smtp_port: {
+          identifier : 'smtp_port',
+          rules      : [
+            {
+              type   : 'empty',
+              prompt : this.i18n.t('Please enter the SMTP port number')
+            },
+            {
+              type   : 'integer',
```
Comment (niranjan94) [inline app/components/forms/admin/settings/system-form.js]: Add `type='number'` to the input too

File: app/components/forms/admin/settings/system-form.js [hunk 3]
```diff
@@ -1,8 +1,186 @@
...
+              prompt : this.i18n.t('Please enter the token for Sendgrid')
+            }
+          ]
+        },
+
+        super_admin_email: {
+          identifier : 'super_admin_email',
+          rules      : [
+            {
+              type   : 'email',
```
Comment (niranjan94) [inline app/components/forms/admin/settings/system-form.js]: Add `type='email'` to the input too

**Comment by CosmicCoder96** [comment]
@niranjan94 @geekyd 
I have made the changes
Please review

**Pull request merged**